src/components/sketches/Scene4.ts

```diff
@@ -0,0 +1,170 @@
+import p5 from 'p5';
+import {
+  BUBBLE_FREQUENCY,
+  BUBBLE_FREQUENCY_ON_LINES,
+  TRANSPARENCY,
+  TRANSPARENCY_RADIUS,
+  INC,
+  INC_START,
+  MAG_INC,
+  SCL,
+  NUM_PARTICLES,
+  STROKE_WEIGHT,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+
+  let start = 0;
+  let cols: number;
+  let rows: number;
+  let zoff = 0;
+  const particles: Particle[] = [];
+  let flowfield: p5.Vector[];
+  let magOff = 0;
+  const showField = false;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+    p.pixelDensity(1);
+    cols = p.floor(window.innerWidth / SCL);
+    rows = p.floor(window.innerHeight / SCL);
+    p.background(0);
+
+    for (let i = 0; i < NUM_PARTICLES; i++) {
+      particles[i] = new Particle();
+    }
+
+    flowfield = new Array(rows * cols);
+
+  };
+
+  p.draw = () => {
+    if (showField) {
+      p.background(0);
+    } else {
+      p.background(p.color(0, 0, 0, 5));
+    }
+    let yoff = start;
+    for (let y = 0; y < rows; y++) {
+      let xoff = start;
+      for (let x = 0; x < cols; x++) {
+        const index = x + y * cols;
+        const angle = p.noise(xoff, yoff, zoff) * p.TWO_PI;
+        const v = p5.Vector.fromAngle(angle); // vector from angle
+        const m = p.map(p.noise(xoff, yoff, magOff), 0, 1, -5, 5);
+        v.setMag(m);
+        if (showField) {
+          p.push();
+          p.stroke(255);
+          p.translate(x * SCL, y * SCL);
+          p.rotate(v.heading());
+          const endpoint = p.abs(m) * SCL;
+          p.line(0, 0, endpoint, 0);
+          if (m < 0) {
+            p.stroke('red');
+          } else {
+            p.stroke('green');
+          }
+          p.line(endpoint - 2, 0, endpoint, 0);
+          p.pop();
+        }
+        flowfield[index] = v;
+        xoff += INC;
+      }
+      yoff += INC;
+    }
+    magOff += MAG_INC;
+    zoff += INC_START;
+    start -= MAG_INC;
+
+    if (!showField) {
```

Comment:
nit:

```
if (showField) return;
... rest of your implementation NOT in a nested If statement
```

Reply:
generally this is seen as more readable because you don't end up having a ton of nested loops

Reply:
deleted showfield, not relevant for the code.